src/Analyser/Scope.php

```diff
@@ -1797,6 +1768,32 @@ private function calculateFromScalars(Expr $node, ConstantScalarType $leftType,
 		return new MixedType();
 	}
 
+	private function resolveExactName(Name $name): ?string
```

Comment:
What's the difference between this and `resolveName`? Could we merge them? (I'm fine if there's an object with `parent` as class name created outside of class, other rules will catch this.)

Reply:
Alright, I'll try to merge them later, the fact that `resolveName()` resolves `static` probably isn't ideal.